dspace-server-webapp/src/test/java/org/dspace/app/rest/GroupRestRepositoryIT.java

```diff
@@ -1538,4 +1650,342 @@ public void removeMemberUnprocessableTest() throws Exception {
             }
         }
     }
+
+    @Test
+    public void deleteGroupTest() throws Exception {
+
+        GroupService groupService = EPersonServiceFactory.getInstance().getGroupService();
+
+        Group parentGroup = null;
+
+        try {
+            context.turnOffAuthorisationSystem();
+
+            parentGroup = groupService.create(context);
+
+            context.commit();
+
+            parentGroup = context.reloadEntity(parentGroup);
+
+            String authToken = getAuthToken(admin.getEmail(), password);
+
+            getClient(authToken).perform(
+                    get("/api/eperson/groups/" + parentGroup.getID())
+            ).andExpect(status().isOk());
+
+            getClient(authToken).perform(
+                    delete("/api/eperson/groups/" + parentGroup.getID())
+            ).andExpect(status().isNoContent());
+
+            getClient(authToken).perform(
+                    get("/api/eperson/groups/" + parentGroup.getID())
+            ).andExpect(status().isNotFound());
+
+        } finally {
+            if (parentGroup != null) {
+                GroupBuilder.deleteGroup(parentGroup.getID());
+            }
+        }
+    }
+
+    @Test
+    public void deleteGroupUnauthorizedTest() throws Exception {
+
+        GroupService groupService = EPersonServiceFactory.getInstance().getGroupService();
+
+        Group parentGroup = null;
+
+        try {
+            context.turnOffAuthorisationSystem();
+
+            parentGroup = groupService.create(context);
+
+            context.commit();
+
+            parentGroup = context.reloadEntity(parentGroup);
```

Comment:
Same as above. should use GroupBuilder, missing a context.restoreAuthSystemState(); and same question about commit() and reloadEntity() as above.